src/webview/VSCodeKaotoEditorChannelApi.ts

```diff
@@ -1,8 +1,20 @@
+import { KaotoEditorChannelApi } from '@kaoto/kaoto';
+import { ISettingsModel, SettingsModel } from '@kaoto/kaoto/models';
 import { DefaultVsCodeKieEditorChannelApiImpl } from '@kie-tools-core/vscode-extension/dist/DefaultVsCodeKieEditorChannelApiImpl';
 import * as vscode from 'vscode';
 
-export class VSCodeKaotoEditorChannelApi extends DefaultVsCodeKieEditorChannelApiImpl {
-    async getCatalogURL(): Promise<string | undefined> {
-        return await vscode.workspace.getConfiguration('kaoto').get('catalog.url');
-    }
+export class VSCodeKaotoEditorChannelApi
+  extends DefaultVsCodeKieEditorChannelApiImpl
+  implements KaotoEditorChannelApi
+{
+  async getCatalogURL(): Promise<string | undefined> {
+    return await vscode.workspace.getConfiguration('kaoto').get('catalog.url');
+  }
+
+  async getVSCodeKaotoSettings(): Promise<ISettingsModel> {
+    const catalogUrl =
+      vscode.workspace.getConfiguration('kaoto').get<string | null>('catalog.url') ?? '';
+
+    return new SettingsModel({ catalogUrl });
```

Comment:
Should it be needed to specify that the value is assigned to the catalogUrl is we do not want to break the compilation when another value will be added?

(it will be more explicit too)

```
return new SettingsModel({ catalogUrl: catalogUrl });
```

Reply:
This `{ catalogUrl: catalogUrl }` is the same as `{ catalogUrl  }`, also the latter is a well-known JS pattern, would you be open to initialize the settings object and then provide it to the `SettingsModel` constructor?

```javascript
const settingsModel: ISettingsModel = {
  catalogUrl: vscode.workspace.getConfiguration('kaoto').get<string | null>('catalog.url') ?? '',
};

return new SettingsModel(settingsModel);
```